tests/Integration/ORM/EntityRelationship/EntityFactoryRelationshipTestCase.php

```diff
@@ -712,6 +713,44 @@ static function(Address $a): void {
         self::assertNotNull($address->getContact());
     }
 
+    /** @test */
+    #[Test]
+    #[DataProvider('provideCascadeRelationshipsCombinations')]
+    #[UsingRelationships(Contact::class, ['category'])]
+    public function find_or_create_used_in_many_actually_create_only_one_object(): void
+    {
+        static::contactFactory()
+            ->many(2)
+            ->create(['category' => lazy(fn() => static::categoryFactory()::findOrCreate([]))]);
+
+        static::contactFactory()::assert()->count(2);
+        static::categoryFactory()::assert()->count(1);
+    }
+
+    /** @test */
+    #[Test]
+    #[DataProvider('provideCascadeRelationshipsCombinations')]
+    #[UsingRelationships(Contact::class, ['category'])]
+    public function ensure_inverse_one_to_many_is_hydrated_before_persisting_in_flush_after(): void
```

Comment:
I'm not 100% sure the name of the test is accurate, this edge case is pretty specific :see_no_evil: (fixes #911)